### PR TITLE
Repo updates

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
fixes

- dev dependencies
- `lerna` deprecated keys
- errors on `npm run lint`
- minimum node version (6.4.0)
- travis test node 12

### CODE REVIEW

**Comment by PopGoesTheWza** [comment]
@koresar I do not think the failing `npm run test` is my doing. I managed to fix a bug in `merge` though.

**Comment by PopGoesTheWza** [comment]
@koresar Tada! All tests passed.

Please review my quick fixes and advise for a cleaner solution.

File: packages/core/merge.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ function mergeOne(dst, src) {
     // is this a regular property?
     if (desc.hasOwnProperty('value')) { // eslint-disable-line
       // Do not merge properties with the 'undefined' value.
-      if (desc.value === undefined) return;
+      if (desc.value === undefined) return dst;
```
Comment (koresar) [inline packages/core/merge.js]: Everything is good with this PR. But this line.

The `merge` algorithm is probably the most difficult part of stampit. I would need to take a very deep look at it before approving the PR. But I don't have time (my newborn is taking over me).

What you can do is go step by step:
* Remove the `.js` file changes from this PR. Leave everything else you've updated.
* I'll merge this PR.
* Submit another PR with `.js` file changes for further review.

Comment (PopGoesTheWza) [reply packages/core/merge.js]: your wish is my command

**Comment by PopGoesTheWza** [comment]
My understanding of what goes on in mergeOne is that when src is undefined, dst is lost.

**Comment by koresar** [comment]
You might have fixed a bug here btw. However, I'd keep such changes
separate.

**Comment by PopGoesTheWza** [comment]
Do you want me to also revert the “assign() more than two parameters” fix?

**Comment by koresar** [comment]
Yes. I meant everything in the JS files.

**Comment by PopGoesTheWza** [comment]
The only .js still modified in this PR is strictly for a linting issue and no code was changed

**Comment by koresar** [comment]
Could you please create branches directly in this repo rather thank forking? I would be able to contribute to your branch with my code.

**Comment by PopGoesTheWza** [comment]
> Could you please create branches directly in this repo rather thank forking? I would be able to contribute to your branch with my code.

@koresar Sure. But currently I can only create branches on `stampit-org/stampit` and not on `stampit-org/stamp` (or I missed something?)

**Comment by koresar** [comment]
Oh. Sorry. I'll double check the permissions etc.

**Pull request merged**